src/StructureFactors.jl

```diff
@@ -931,3 +931,75 @@ function sf_slice(sf::StructureFactor, points::Vector;
     return slice
 end
 
+
+
+#========== Powder Averaging WIP ==========#
+""" gaussian_smoothing(k, q, S, L)
+    Returns 'Powder-Averaged' contribution to different bins according to 
+the exact formula for Gaussian Smoothing Kernel for each S(q). 
+
+Args 
+    k :: Range of 'bins' for Powder Averaging. 
+    q :: Magnitude of 'q' vector; Gaussian smoothing only a function of magnitude. 
+    S :: Intensity at 'q' vector. 
+    L :: Length-scale for smoothing kernel; suggested maximum(Lx, Ly, Lz). 
+        User can control this but issues may arise if σ for Gaussian Smoothing kernel is too big or small w.r.t spacing between q-vectors. 
+"""
+function gaussian_smoothing(k, q, S, L)
+    sig2 = (2π/L)^2.0 
+    c = (2π*sig2)^(3/2)
+    exp_pos = exp.( (.-k.*k.-q.*q.+2.0.*k.*q) ./ (2.0.*sig2) )
```

Comment:
I think all the dots can go away with broadcasting macro:
```julia
exp_pos = @. exp( (-k*k-q*q+2k*q) / (2sig2) )
```

Reply:
As we discussed on Slack, another option is:
```julia
exp_pos = exp( (-k.*k-q.*q+2k.*q) / 2sig2 )
```
Note that if you want to write into a preallocated array `exp_pos`, then the idiomatic code would be:
```julia
@. exp_pos = exp( (-k*k-q*q+2k*q) / 2sig2 )
```
with the `@.` macro in front of the entire line.